### PR TITLE
test: add cert-manager module test, add github actions test workflow

### PR DESCRIPTION
**Pull request by mrparkers**

### CODE REVIEW

File: .github/workflows/destroy-vcluster.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: "Destroy vcluster"
+
+on:
+  pull_request:
+    types:
+      - closed
+  push:
+    branches:
+      - master
```
Comment (ChrisSchreiber) [inline .github/workflows/destroy-vcluster.yml:9]: will this make it run twice (on pr close and merge to master)?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+          cd lead-terraform
+          go test -v ./modules/./...
+
+      - name: Destroy vcluster
+        if: github.ref == 'refs/heads/master'
+        env:
+          VCLUSTER_NAME: ${{ steps.vcluster-name.outputs.name }}
+          TERRAGRUNT_IAM_ROLE: ${{ secrets.AWS_LEAD_TERRAFORM_TEST_ROLE }}
+        run: terragrunt destroy --terragrunt-working-dir ./lead-environments/aws/vcluster -auto-approve
```
Comment (ChrisSchreiber) [inline .github/workflows/test.yml:91]: Do we still need this with the other workflow?

Comment (mrparkers) [reply .github/workflows/test.yml:91]: Yes, this is so the vcluster that gets created for the run against `master` gets cleaned up.

File: .github/workflows/destroy-vcluster.yml (hunk 1 above)
Comment (mrparkers) [reply .github/workflows/destroy-vcluster.yml:9]: I messed this up, I meant to only run this on PR close

**Review by ChrisSchreiber — APPROVED** [review]

**Pull request merged**